Automattic/jetpack
# Blaze: move plugin out of the Jetpack plugin and into package

#### Changes proposed in this Pull Request:

This is a follow-up to the following PRs:

- #27928 introducing the panel
- #27960 changing the panel's design.

This commit brings in:
- The panel's code from the Jetpack plugin.
    - We change the way we register the panel, so it can be registered outside of the Jetpack plugin; it shouldn't rely on the list of enabled extensions that is managed by the Jetpack plugin.
- The tooling to build the JavaScript bundle in the package.
- The dependencies for it.
- We then enqueue that bundle in the block editor.

#### Other information:


#### Jetpack product discussion

* 1386-gh-dotcom-forge

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

`jetpack build packages/promote-posts` 
`jetpack watch packages/promote-posts`

* Start on a WordPress.com simple or WoA site that's connected to WordPress.com.
* Switch to this branch, or use the Jetpack Beta plugin to run this branch.
* Add the following in a place that's going to be loaded:
```php
use Automattic\Jetpack\Promote_Posts;
Promote_Posts::configure();
```
* Go to Posts > Add New
* Publish a post.
* Check the post-publish sidebar panel
* Is the button there?
